corehq/apps/reports/tasks.py

```diff
@@ -163,33 +167,36 @@ def apps_update_calculated_properties():
         send_to_elasticsearch('apps', doc, es_merge_update=True)
 
 
-@task(serializer='pickle', ignore_result=True)
-def export_all_rows_task(ReportClass, report_state, recipient_list=None, subject=None):
-    report = object.__new__(ReportClass)
-    report.__setstate__(report_state)
-    report.rendered_as = 'export'
-
-    setattr(report.request, 'REQUEST', {})
-    file = report.excel_response
-    report_class = report.__class__.__module__ + '.' + report.__class__.__name__
-
-    if report.domain is None:
+@task(ignore_result=True)
+def export_all_rows_task(user_id, report_class, domain, slug, name,
+                         export_format, export_table_parts, recipient_list=None, subject=None):
+    file = io.BytesIO()
+    table = chain(export_table_parts[1], export_table_parts[2])
+    if export_table_parts[3]:
+        table = chain(table, export_table_parts[3])
+    if export_table_parts[4]:
+        table = chain(table, export_table_parts[4])
+    export_table = [[export_table_parts[0], table]]
```

Comment:
Based on how you are using the indices, I think a more readable/appropriate structure would be to use a dictionary instead of a list. Something like `if 'total_row' in json_report:` is more intuitive than `if json_report[3]:`. Plus, if `json_report` did not contain at least 5 entries for whatever reason, this would raise an exception. 

Also, more of a nit, but similar to my previous naming comment, I'd prefer a different name than `export_table_parts`. My main issue is it sounds like a verb/action, not a variable. I suggested `json_report` because that is my understanding of what this is, but feel free to choose something else.